sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -174,6 +184,50 @@ private async Task<HttpResponseMessage> ProcessEventsAsync(JArray events, string
             return new HttpResponseMessage(HttpStatusCode.Accepted);
         }
 
+        private async Task<FunctionResult> ExecuteWithTracingAsync(string functionName, TriggeredFunctionData triggerData)
+        {
+            using DiagnosticScope scope = _diagnosticScopeFactory.CreateScope(DiagnosticScopeName, DiagnosticScope.ActivityKind.Consumer);
+            if (scope.IsEnabled)
+            {
+                if (triggerData.TriggerValue is JArray evntArray)
+                {
+                    foreach (JToken eventToken in evntArray)
+                    {
+                        AddLinkIfEventHasContext(scope, eventToken);
+                    }
+                }
+                else if (triggerData.TriggerValue is JToken eventToken)
+                {
+                    AddLinkIfEventHasContext(scope, eventToken);
+                }
+            }
+
+            scope.Start();
+
+            FunctionResult result = await _listeners[functionName].Executor.TryExecuteAsync(triggerData, CancellationToken.None).ConfigureAwait(false);
```

Comment:
In original code we did not start the execution but just added a new task to `executions` collections and then call `await Task.WhenAll(executions).ConfigureAwait(false);` It looks like the new code will give consistently executions instead of parallel for batch scenario.

Reply:
ExecuteWithTracingAsync isn't awaited, so it should be the same.